src/controller/CHIPDeviceController.h

```diff
@@ -957,14 +962,19 @@ class DLL_EXPORT DeviceCommissioner : public DeviceController,
     CommissioneeDeviceProxy * FindCommissioneeDevice(const Transport::PeerAddress & peerAddress);
     void ReleaseCommissioneeDevice(CommissioneeDeviceProxy * device);
 
+    bool ExtendArmFailSafeInternal(DeviceProxy * proxy, CommissioningStage step, uint16_t armFailSafeTimeout,
```

Comment:
This needs some documentation, especially because `useContext` is not very clear.

Perhaps that argument should be called `isSequentialCommissioningStep` or something?

I would also prefer the new argument not be optional, since forgetting it when it should be false can lead to VerifyOrDie failures.  Just require callers to think about what they want to pass.